diff --git a/Makefile b/Makefile
index cfa081b7..47790d09 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@ SHELL = /bin/bash
 
 include Makefile.e2e
 
-REGISTRY = kubeovn
+REGISTRY = r.awan.app/library/kube-ovn
 DEV_TAG = dev
 RELEASE_TAG = $(shell cat VERSION)
 VERSION = $(shell echo $${VERSION:-$(RELEASE_TAG)})
@@ -46,6 +46,11 @@ build-go-arm:
 	CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go build -buildmode=pie -o $(CURDIR)/dist/images/kube-ovn-cmd -ldflags $(GOLDFLAGS) -v ./cmd
 	CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go build -buildmode=pie -o $(CURDIR)/dist/images/kube-ovn-webhook -ldflags $(GOLDFLAGS) -v ./cmd/webhook
 
+.PHONY: build-go-ppc64le
+build-go-ppc64le:
+	CGO_ENABLED=0 GOOS=linux GOARCH=ppc64le go build -buildmode=pie -o $(CURDIR)/dist/images/kube-ovn-cmd -ldflags $(GOLDFLAGS) -v ./cmd
+	CGO_ENABLED=0 GOOS=linux GOARCH=ppc64le go build -buildmode=pie -o $(CURDIR)/dist/images/kube-ovn-webhook -ldflags $(GOLDFLAGS) -v ./cmd/webhook
+
 .PHONY: build-dev
 build-dev: build-go
 	docker build --build-arg ARCH=amd64 -t $(REGISTRY)/kube-ovn:$(DEV_TAG) -f dist/images/Dockerfile dist/images/
@@ -67,6 +72,10 @@ base-amd64-dpdk:
 base-arm64:
 	docker buildx build --platform linux/arm64 --build-arg ARCH=arm64 -t $(REGISTRY)/kube-ovn-base:$(RELEASE_TAG)-arm64 -o type=docker -f dist/images/Dockerfile.base dist/images/
 
+.PHONY: base-ppc64le
+base-ppc64le:
+	docker buildx build --platform linux/ppc64le --build-arg ARCH=ppc64le -t $(REGISTRY)/kube-ovn-base:$(RELEASE_TAG)-ppc64le -o type=docker -f dist/images/Dockerfile.base dist/images/
+
 .PHONY: image-kube-ovn
 image-kube-ovn: build-go
 	docker buildx build --platform linux/amd64 --build-arg ARCH=amd64 -t $(REGISTRY)/kube-ovn:$(RELEASE_TAG) -o type=docker -f dist/images/Dockerfile dist/images/
@@ -98,6 +107,11 @@ release-arm: build-go-arm
 	docker buildx build --platform linux/arm64 --build-arg ARCH=arm64 -t $(REGISTRY)/kube-ovn:$(RELEASE_TAG) -o type=docker -f dist/images/Dockerfile dist/images/
 	docker buildx build --platform linux/arm64 --build-arg ARCH=arm64 -t $(REGISTRY)/vpc-nat-gateway:$(RELEASE_TAG) -o type=docker -f dist/images/vpcnatgateway/Dockerfile dist/images/vpcnatgateway
 
+.PHONY: release-ppc64le
+release-ppc64le: build-go-ppc64le
+	docker buildx build --platform linux/ppc64le --build-arg ARCH=ppc64le -t $(REGISTRY)/kube-ovn:$(RELEASE_TAG) -o type=docker -f dist/images/Dockerfile dist/images/
+	docker buildx build --platform linux/ppc64le --build-arg ARCH=ppc64le -t $(REGISTRY)/vpc-nat-gateway:$(RELEASE_TAG) -o type=docker -f dist/images/vpcnatgateway/Dockerfile dist/images/vpcnatgateway
+
 .PHONY: push-dev
 push-dev:
 	docker push $(REGISTRY)/kube-ovn:$(DEV_TAG)
diff --git a/dist/images/Dockerfile b/dist/images/Dockerfile
index f3462731..56c1be3d 100644
--- a/dist/images/Dockerfile
+++ b/dist/images/Dockerfile
@@ -1,5 +1,5 @@
 # syntax = docker/dockerfile:experimental
-FROM kubeovn/kube-ovn-base:v1.11.0
+FROM r.awan.app/library/kube-ovn/kube-ovn-base:v1.11.0
 
 COPY *.sh /kube-ovn/
 COPY kubectl-ko /kube-ovn/kubectl-ko
diff --git a/dist/images/Dockerfile.base b/dist/images/Dockerfile.base
index 9dec8edd..d0caf17e 100644
--- a/dist/images/Dockerfile.base
+++ b/dist/images/Dockerfile.base
@@ -38,7 +38,7 @@ RUN cd /usr/src/ && \
 
 RUN dpkg -i /usr/src/python3-openvswitch*.deb /usr/src/libopenvswitch*.deb
 
-RUN cd /usr/src/ && git clone -b branch-22.03 --depth=1 https://github.com/ovn-org/ovn.git && \
+RUN cd /usr/src/ && git clone -b v22.03.2 --depth=1 https://github.com/ovn-org/ovn.git && \
     cd ovn && \
     # fix ssl listen address
     curl -s https://github.com/kubeovn/ovn/commit/7992d55af9ac356d3ad934764f473d1b6aa4cf64.patch | git apply && \
@@ -105,7 +105,7 @@ RUN --mount=type=bind,target=/packages,from=ovs-builder,source=/packages  \
     rm -rf /var/lib/openvswitch/pki/
 
 ENV DUMB_INIT_VERSION="1.2.5"
-RUN dump_arch="x86_64"; \
+RUN dump_arch=${ARCH}; \
     if [ "$ARCH" = "arm64" ]; then dump_arch="aarch64"; fi; \
     curl -sSf -L --retry 5 -o /usr/bin/dumb-init https://github.com/Yelp/dumb-init/releases/download/v${DUMB_INIT_VERSION}/dumb-init_${DUMB_INIT_VERSION}_${dump_arch} && \
     chmod +x /usr/bin/dumb-init
diff --git a/kubeovn-helm/values.yaml b/kubeovn-helm/values.yaml
index 9bbb1981..d96d8f74 100644
--- a/kubeovn-helm/values.yaml
+++ b/kubeovn-helm/values.yaml
@@ -3,7 +3,7 @@
 # Declare variables to be passed into your templates.
 global:
   registry:
-    address: kubeovn
+    address: r.awan.app/library/kube-ovn
     imagePullSecrets: []
   images:
     kubeovn:
